Formula/synergy-core.rb

```diff
@@ -0,0 +1,376 @@
+
+    def kill_and_wait_for(process)
+      Process.kill("KILL", process.pid)
+      Process.wait(process.pid)
+    end
+
+    # In certain cases (discussed below), it is not possible to do any nontrivial
+    # tests. To make sure that Synergy at least compiled successfully, we'll do
+    # some totally trivial checks here. These checks are bad, but they are better
+    # than nothing, which is what would otherwise be possible in certain cases.
+    assert_equal `#{opt_bin}/synergys`, "synergys: no configuration available\n"
+    assert_equal `#{opt_bin}/synergyc`.split("\n")[0], "synergyc: a server address or name is required"
+    synergys_version = `#{opt_bin}/synergys --version`.split("\n")[0]
+    synergyc_version = `#{opt_bin}/synergyc --version`.split("\n")[0]
+    if build.head?
+      version_string = Regexp.quote(version.to_s.split("-")[1])
+      assert_match(/synergys [\-.0-9a-z]*-#{version_string}[a-f0-9]*, protocol version/, synergys_version)
+      assert_match(/synergyc [\-.0-9a-z]*-#{version_string}[a-f0-9]*, protocol version/, synergyc_version)
+    else
+      # The version information returned by synergys and synergyc is unusual.
+      # The actual version has four numbers (a.b.c.d), but the `--version` output
+      # omits the fourth number (i.e., it returns a.b.c). Additionally, the `--version`
+      # output appends some other numbers, letters, and symbols that are not easily
+      # recoverable, so we use regexps to deal with that.
+      version_string = Regexp.quote(version.to_s.split(".")[0..2].join("."))
+      assert_match(/synergys #{version_string}[\-.0-9a-z]*, protocol version/, synergys_version)
+      assert_match(/synergyc #{version_string}[\-.0-9a-z]*, protocol version/, synergyc_version)
+    end
+
+    # Spawn Synergy server.
+    IO.popen [
+      opt_bin/"synergys",
+      "-f", # run in foreground
+      "-c", conf_file,
+      "-a", "127.0.0.1:#{port}",
+      "-n", "test-server",
+      err: [:child, :out] # redirect child's stderr to its stdout
+    ] do |io_server|
+      # We'll check for these messages below.
+      log_permission_error = "FATAL: A runtime error occurred: " \
+                             "assistive devices does not trust this process, allow it in system settings.".freeze
+      log_no_display_error = "FATAL: A display invalid exception error occurred: " \
+                             "failed to initialize screen shape".freeze
+
+      # The whole point of the Synergy server is to read input from the user's
+      # mouse and keyboard. As a result, if we are running on MacOS and the
+      # user has not granted the "Accessibility" permission to the process
+      # running this test, the Synergy server will categorically refuse to
+      # start up. Unfortunately, there's nothing we can do about this, but
+      # it's not a flaw in Synergy itself (or in this test). We do verify
+      # that we receive the expected error message (log_permission_error).
```

Comment:
Since we care most about CI I think we should only test the flow where no permissions are granted and there is no screen.

Reply:
We need to be able to handle the test where the application has permissions, particularly because on GNU/Linux, the app will always have permission. There's no simple way to fool Synergy into thinking it doesn't have permission, or there isn't a display, if in fact it does have permission and there is a display. However, I agree with you that we can downplay the less-common possibilities here, so I've simplified the code and removed all the remaining messages, except for the warning about anomalous mouse and keyboard functionality in the case where that is relevant.